include/glow/Base/Tensor.h

```diff
@@ -78,6 +82,13 @@ class Tensor final {
   /// \returns true if it is an unowned tensor.
   bool isUnowned() const { return isUnowned_; }
 
+  /// \returns the size of the unpadded memory region. If unpaddedSize_ is not
+  /// set return the size of the entire payload.
+  size_t getUnpaddedSizeInBytes();
+
+  /// Set the size of unpaddedSize_ to \p size
+  void setUnpaddedSize(size_t size) { unpaddedSize_ = size; }
```

Comment:
What do you think about killing the setter in favor of a new flavor of unowned constructor?  I always like to avoid exposing data members that can be arbitrarily twiddled.